src/main/java/io/cdap/plugin/ariba/source/AribaInputFormat.java

```diff
@@ -0,0 +1,139 @@
+ */
+public class AribaInputFormat extends InputFormat<NullWritable, StructuredRecord> {
+
+  private static final Gson GSON = new Gson();
+  private static final Logger LOG = LoggerFactory.getLogger(AribaInputFormat.class);
+  private final List<InputSplit> resultSplits = new ArrayList<>();
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext) throws IOException {
+    AribaPluginConfig pluginConfig = getPluginConfig(jobContext);
+    AribaServices aribaServices = new AribaServices(pluginConfig);
+    boolean previewEnabled = Boolean.parseBoolean(jobContext.getConfiguration().
+                                                    get(ResourceConstants.IS_PREVIEW_ENABLED));
+    createJob(pluginConfig, aribaServices, previewEnabled, null);
+    return resultSplits;
+  }
+
+  @Override
+  public RecordReader<NullWritable, StructuredRecord> createRecordReader(InputSplit inputSplit, TaskAttemptContext
+    taskAttemptContext) throws IOException {
+    AribaPluginConfig pluginConfig = getPluginConfig(taskAttemptContext);
+    AribaServices aribaServices = new AribaServices(pluginConfig);
+    Schema outputSchema = Schema.parseJson(taskAttemptContext.getConfiguration().get(ResourceConstants.OUTPUT_SCHEMA));
+    return new AribaRecordReader(aribaServices, outputSchema);
+  }
+
+  private AribaPluginConfig getPluginConfig(JobContext taskAttemptContext) {
+    return GSON.fromJson(taskAttemptContext.getConfiguration().
+                           get(ResourceConstants.ARIBA_PLUGIN_PROPERTIES), AribaPluginConfig.class);
+  }
+
+  /**
+   *  call for create Job
+   *
+   * @param pluginConfig  Plugin config
+   * @param aribaServices  Ariba services
+   * @param isPreviewEnabled Is preview enabled
+   * @param pageToken    Page token
+   * @throws IOException
+   */
+  private void createJob(AribaPluginConfig pluginConfig,
+                         AribaServices aribaServices, boolean isPreviewEnabled,
+                         String pageToken) throws IOException {
+    JsonNode jobData;
+    String jobId;
+
+    try {
+      JsonNode createJobResponse = aribaServices.createJob(pluginConfig,
+                                                           aribaServices.getAccessToken(), pageToken);
+      jobId = createJobResponse.get(ResourceConstants.JOB_ID).asText();
+    } catch (AribaException | InterruptedException | IOException exception) {
+      throw new IOException(exception.getMessage(), exception);
+    }
+
+    while (true) {
+      try {
+        LOG.info("Fetching Data For Job Id: {}", jobId);
+        jobData = aribaServices.fetchJobStatus(aribaServices.getAccessToken(), jobId);
+        if (jobData.get(ResourceConstants.STATUS).asText().equals(ResourceConstants.COMPLETED) ||
+          jobData.get(ResourceConstants.STATUS).asText().equals(ResourceConstants.COMPLETED_ZERO_RECORDS)) {
+          break;
+        }
+        Thread.sleep(60000);
```

Comment:
Make this a constant. Also, how did you decide on 60 seconds? How long does a job usually take to complete?

Reply:
Average time is around 7 to 10 minutes, but that's not fixed. So we try every minute whether it's completed or not just to be sure that we don't wait unnecessarily when the job completes.